File: leetcode/easy/remove-one-element-to-make-the-array-strictly-increasing.ts
// https://leetcode.com/problems/remove-one-element-to-make-the-array-strictly-increasing/

// MINE
/*
function canBeIncreasing(nums: number[]): boolean {
    for(let i=0; i<nums.length; ++i) {
        const copy = [...nums]
        copy.splice(i, 1)
        
        let isAsc = true
        for(let j=1; j<copy.length; ++j) {
            if(copy[j-1] >= copy[j]) {
                isAsc = false
                break
            }
        }
        if(isAsc) return true
    }
    return false
};
*/

// ANSWER
function canBeIncreasing(nums: number[]): boolean {
  let count = 0
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] <= nums[i - 1]) {
      count++
      if (i > 1 && nums[i] <= nums[i - 2]) nums[i] = nums[i - 1]
    }
  }
  return count <= 1
}

File: leetcode/easy/remove-one-element-to-make-the-array-strictly-increasing.py
# https://leetcode.com/problems/remove-one-element-to-make-the-array-strictly-increasing/

class Solution:
    # MINE
    def canBeIncreasing(self, nums: List[int]) -> bool:
        count = 0
        for i in range(1, len(nums)):
            if nums[i-1] >= nums[i]:
                count += 1
                if i > 1 and nums[i-2] >= nums[i]:
                    nums[i] = nums[i-1]
        return count <= 1

# range(stop) -> stop-1
# NO ++ operator in python
